src/aalcameraserviceplugin.cpp

```diff
@@ -96,38 +86,63 @@ QString AalServicePlugin::deviceDescription(const QByteArray &service, const QBy
     }
 }
 
+// krillin / vegetahd lies to us - the top of the front facing camera
+// points to the right of the screen (viewed from the front), which means
+// the camera image needs rotating by 270deg with the device in its natural
+// orientation (portrait). It tells us the camera orientation is 90deg
+// though (see https://launchpad.net/bugs/1567542)
+// https://git.launchpad.net/oxide/tree/shared/browser/media/oxide_video_capture_device_hybris.cc#n92
+
+// Meanwhile, cooler (M10 HD) gives the orientation of 0 for all cameras while
+// actually has its cameras pointing toward the "bottom" of the device.
+// Except, both the screen and the orientation sensor agrees that the native
+// orientation is "portrait", not the apparent "landscape". This means the
+// back camera's returned orientation must be 270 deg (Qt convention), while
+// the front one's must be 90.
+
+// This map contains all overrides we have. The format for the key is
+// "<device codename>_<camera_id>" where the camera id is usually "0" for back
+// facing camera and "1" for front facing one. The value contains the orientation
+// we would return (in Qt convention i.e. no conversion required).
+static const QHash<QString, int> cameraOrientationOverride = {
+    {"krillin_1", 90},
+    {"vegetahd_1", 90},
+    {"cooler_0", 270},
+    {"cooler_1", 90},
```

Comment:
Would refactoring this to a slightly deeper data structure make more sense? For example:

```
{
  "krillin": [90, 90],
  "cooler": [270, 90]
}
```

So to get the front-facing camera you do the equivalent of `cameraOrientationOverride["krillin"][0]`? Makes the override key easier to reason about.

Alternatively, do we want to load this information from a config file? This would make it easier for people porting to new devices who have these ridiculous overrides, they won't need to make changes to some C++ for a relatively simple override.

We're not supposed to use 'ubuntu' for package or config file names any more, so maybe something like `/etc/qtunityclient/camera/` would make more sense. Would basically resolve ourselves to renaming QtUbuntu, `QtUnityClient`.